script/rewrite_policy_publication_history.rb

```diff
@@ -4,27 +4,44 @@
 # NOTE: This script is NOT idempotent and so should only be run once!
 require 'csv'
 
-mapping_csv_path = Rails.root+"lib/tasks/election/2015-04-20-13-42-38-policy_paper_creation_output.csv"
 logger = Logger.new(STDOUT)
 
-# Make sure no publications have an open edition as we can't handle those
-CSV.parse(File.open(mapping_csv_path).read, headers: true).each do |row|
-  publication = Publication.find_by_id(row["policy_paper_id"])
-  next unless publication
+def with_publications_and_policies(&block)
```

Comment:
What's the reasoning behind multiple runs through this loop now? Seems inefficient.